chain/txValidator.go

```diff
@@ -145,12 +145,16 @@ func CheckTransactionPayload(txn *transaction.Transaction, height uint32) error
 			return errors.New("Register name transaction is not supported yet")
 		}
 		pld := payload.(*pb.RegisterName)
-		match, err := regexp.MatchString("(^[A-Za-z][A-Za-z0-9-_.+]{2,254}$)", pld.Name)
+		regexPattern := "(^[A-Za-z][A-Za-z0-9+]{5,254})\\.nkn$"
+		if height < config.OldNameRegistrationHeight {
+			regexPattern = "(^[A-Za-z][A-Za-z0-9-_.+]{2,254}$)"
+		}
+		match, err := regexp.MatchString(regexPattern, pld.Name)
 		if err != nil {
 			return err
 		}
 		if !match {
-			return fmt.Errorf("name %s should start with a letter, contain A-Za-z0-9-_.+ and have length 3-255", pld.Name)
+			return fmt.Errorf("name %s should start with a letter, contain A-Za-z0-9 with .nkn suffix and have length 6-255", pld.Name)
```

Comment:
Since regex is height dependent now, we can simply return regexPattern in error, like `fmt.Errorf("name should match regex %s", regexPattern)`